persistent/_compat.py

```diff
@@ -35,3 +48,179 @@
     PYTHON2 = True
 
     intern = intern
+
+PYPY = hasattr(sys, 'pypy_version_info')
+
+
+def _c_optimizations_required():
+    """
+    Return a true value if the C optimizations are required.
+
+    This uses the ``PURE_PYTHON`` variable as documented in `_use_c_impl`.
+    """
+    pure_env = os.environ.get('PURE_PYTHON')
+    require_c = pure_env == "0"
+    return require_c
+
+
+def _c_optimizations_available():
+    """
+    Return the C optimization modules, if available, otherwise
+    a false value.
+
+    If the optimizations are required but not available, this
+    raises the ImportError. Either all optimization modules are
+    available or none are.
+
+    This does not say whether they should be used or not.
+    """
+    catch = () if _c_optimizations_required() else (ImportError,)
+    try:
+        from persistent import cPersistence
+        from persistent import cPickleCache
+        from persistent import _timestamp
+        return {
+            'persistent.persistence': cPersistence,
+            'persistent.picklecache': cPickleCache,
+            'persistent.timestamp': _timestamp,
+        }
+    except catch: # pragma: no cover (only Jython doesn't build extensions)
+        return {}
+
+
+def _c_optimizations_ignored():
+    """
+    The opposite of `_c_optimizations_required`.
+    """
+    pure_env = os.environ.get('PURE_PYTHON')
+    # The extensions can be compiled with PyPy 7.3, but they don't work.
+    return PYPY or (pure_env is not None and pure_env != "0")
```

Comment:
So setting PURE_PYTHON to an empty string is equivalent to setting it to 1?  I'm not sure how I feel about that, but okay.

Reply:
I was trying to preserve compatibility. Putting PURE_PYTHON in the environment *period* would result in no C extensions; the value was immaterial. 

Except as I write that, I realize that no, this package actually handled things a bit differently than, say, [zope.proxy](https://github.com/zopefoundation/zope.proxy/blob/5773b8b33e42d28962684f7647f4dca089518259/src/zope/proxy/__init__.py#L515). Instead of checking for its presence *in* the environment, it checks its value. 

https://github.com/zopefoundation/persistent/blob/29d207e7849bb55b371e1a4c6674eb4264ab9fbd/persistent/_compat.py#L18

https://github.com/zopefoundation/persistent/blob/29d207e7849bb55b371e1a4c6674eb4264ab9fbd/persistent/__init__.py#L55-L56

So maybe that can/should be simplified/corrected.